image/manifest_test.go

```diff
@@ -66,58 +54,33 @@ func TestUnpackLayerDuplicateEntries(t *testing.T) {
 }
 
 func TestUnpackLayer(t *testing.T) {
-	tmp1, err := ioutil.TempDir("", "test-layer")
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer os.RemoveAll(tmp1)
-	err = os.MkdirAll(filepath.Join(tmp1, "blobs", "sha256"), 0700)
-	if err != nil {
-		t.Fatal(err)
-	}
-	tarfile := filepath.Join(tmp1, "blobs", "sha256", "test.tar")
-	f, err := os.Create(tarfile)
+	root, err := ioutil.TempDir("", "test-layer")
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer os.RemoveAll(root)
 
-	gw := gzip.NewWriter(f)
-	tw := tar.NewWriter(gw)
-
-	tw.WriteHeader(&tar.Header{Name: "test", Size: 4, Mode: 0600})
-	io.Copy(tw, bytes.NewReader([]byte("test")))
-	tw.Close()
-	gw.Close()
-	f.Close()
-
-	// generate sha256 hash
-	h := sha256.New()
-	file, err := os.Open(tarfile)
+	err = os.MkdirAll(filepath.Join(root, "blobs", "sha256"), 0700)
```

Comment:
It seems some inappropriate to use `createImageLayoutBundle` here.
the `TestUnpackLayer` prerequisite is just layer file with right path and its descriptor. It even needn't real manifest file. So it is enough to make layer dir and create layer file.
If `TestUnpackLayer` function is missing some test case( I read `testManifest.unpack()` function, and think `TestUnpackLayer` is fine), it is another story.
if it is correct?